addons/web/static/src/legacy/js/widgets/colorpicker.js

```diff
@@ -124,6 +124,19 @@ var ColorpickerWidget = Widget.extend({
      */
     setSelectedColor: function (color) {
         var rgba = ColorpickerWidget.convertCSSColorToRgba(color);
+        if ((color == 'black') 
+            || (color == 'white')
+            || (color.match(/^[1-9]00$/)) 
+            || (color.match(/black-\d{2}/)) 
+            || (color.match(/white-\d{2}/)) 
+            || (color.match(/o-color-\d{1}/)) 
+        ) {
+            const exampleEl = document.querySelector('.bg-'.concat(color));
+            if (exampleEl) {
+                const exampleElStyle = getComputedStyle(exampleEl);
+                rgba = ColorpickerWidget.convertCSSColorToRgba(exampleElStyle.backgroundColor);
+            }
+        }
```

Comment:
This is not good enough for multiple reasons:
1. If I am not mistaken, we have a "colorpicker" and a "colorpalette" component. The colorpalette one contains an inner colorpicker to choose custom colors. From the "colorpicker" point of view, you should not have to deal with color CSS classes and their related colors... unless maybe it was given as a config from the parent colorpalette. In this case, I think the parent should just be adapted: when it calls`setSelectorColor` of its inner colorpicker, instead of sending a named color, it should first process it to a real CSS color that the colorpicker can understand.
2. `document.querySelector` to find information about color is never the right answer. The UI should never have to search its own DOM to find information like that. The colorpalette can probably match each named color to the actual CSS color using `getCSSVariableValue` at worst.
3. Beware that regex, while great, are costly operations: avoid 4 `match` in a row. Also we don't want to match thing with regexes like that: colorpalettes are defined with a given set of colors: this should just have been about `givenSetsOfColors.include(color)` instead of trying to understand the name. But as explained (point 1 & 2), it should just be done on the colorpalette side with something like "get CSS color from the name. No color? Then try to extract color rgb values directly. Still nothing? Then it's something unknown, don't update the UI.